packages/fetch-and-compile/test/fetch.test.ts

```diff
@@ -0,0 +1,89 @@
+import assert from "assert";
+import "mocha";
+import Config from "@truffle/config";
+import { fetchAndCompile } from "../lib/index";
+import axios from "axios";
+import sinon from "sinon";
+const fixture: any = require("./fixture.js");
+
+function stubAxiosGetMethod (url: string, address: string, data: object) {
+  sinon.stub(axios, 'get').withArgs(url, {
+    params: {
+      module: "contract",
+      action: "getsourcecode",
+      address: address,
+      apikey: ''
+    },
+    responseType: "json",
+    maxRedirects: 50
+  }).returns(Promise.resolve({ data: data }));
+};
+
+afterEach(function (){
+  //TS can't detect that is a sinon stub so we have to use ts-ignore
+  //@ts-ignore
+  //restoring stub
+  axios.get.restore();
+});
+
+describe("fetchAndCompile", function () {
+  it('verifes contract from mainnet', async function () {
+    const config = Config.default().merge({
+      networks: {
+        mainnet: {
+          network_id: 1
+        }
+      },
+      network: "mainnet",
+    });
+    const address = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D';
+    //asserting that mainnet url and contract address is passed as args
+    stubAxiosGetMethod("https://api.etherscan.io/api", address, fixture.mainnetData);
+    //TS can't detect that is a sinon stub so we have to use ts-ignore
+    //@ts-ignore
+    axios.get.callThrough();
+    const result = await fetchAndCompile(address, config);
+    let contractName = result.sourceInfo.contractName;
```

Comment:
Btw, I only just noticed you're getting `contractName` out of `sourceInfo`.  That tests the fetching part, but not so much the compiling part (other than that the compiling part doesn't crash).  I think this test (and the others) would be better if, in addition to testing the contract name found in `sourceInfo`, it also tested the contract name found in `compilations`.

Reply:
Btw @leeftk I don't know if you've merged in or rebased against `develop` since #4447 was merged, but if so you might want to add tests of that as well. :)  Or if you don't want to do that then I can get it as follow-on work.

Reply:
Great catch! I didn't even think that about this until you mentioned it. I added an assertion that also tests the `compilations` part of the function. Let me know if what I added will suffice.

Reply:
Cool, I see you've done this!  I have some comments on that but those will be separate. :)